src/qt3utils/datagenerators/spectrometers/spectrometer.py

```diff
@@ -0,0 +1,230 @@
+        """
+        pass
+
+    @property
+    def clock_rate(self) -> float:
+        """
+        The clock rate of a single exposure (1/exposure_time in Hz).
+        """
+        try:
+            _t = self.exposure_time / 1000.0  # Converting from milliseconds to seconds.
+        except Exception as e:
+            self.logger.error(e)
+            _t = 2
+        return 1.0 / _t
+
+    def acquire(
+            self,
+            acquisition_mode: Literal['single', 'step-and-glue', 'kinetic series', 'accumulation'],
+            **kwargs
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        General method that acquires data depending on the acquisition mode.
+        If you want to add a new mode, overwrite this method, call it via super().acquire(...) and then
+        use an if statement for each different mode you implemented.
+
+        Parameters
+        ----------
+        acquisition_mode : Literal['single', 'step-and-glue', 'kinetic series', 'accumulation']
+            The acquisition mode.
+        **kwargs
+            Additional keyword arguments.
+        Returns
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            The single or multiple spectra and the wavelength array.
+        """
+        if acquisition_mode not in self.ACQUISITION_MODES:
+            raise ValueError(f'Unsupported acquisition mode: {acquisition_mode}')
+        if acquisition_mode == 'single':
+            valid_kwargs = filter_only_valid_kwargs_for_method(self.single_acquisition, kwargs)
+            return self.single_acquisition(**valid_kwargs)
+        if acquisition_mode == 'step-and-glue':
+            valid_kwargs = filter_only_valid_kwargs_for_method(self.step_and_glue_acquisition, kwargs)
+            return self.step_and_glue_acquisition(**valid_kwargs)
+        if acquisition_mode == 'kinetic series':
+            valid_kwargs = filter_only_valid_kwargs_for_method(self.kinetic_series_acquisition, kwargs)
+            return self.kinetic_series_acquisition(**valid_kwargs)
+        if acquisition_mode == 'accumulation':
+            valid_kwargs = filter_only_valid_kwargs_for_method(self.accumulation_acquisition, kwargs)
+            return self.accumulation_acquisition(**valid_kwargs)
+
+    def single_acquisition(self, **kwargs) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Method that acquires a single frame.
+
+        Returns
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            The single-frame spectrum and the wavelength array.
+        """
+        raise NotImplementedError()
+
+    def step_and_glue_acquisition(self, **kwargs) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Method that acquires a single frame via the step-and-glue method.
+
+        Returns
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            The step-and-glued spectrum and the wavelength array.
+        """
+        raise NotImplementedError()
```

Comment:
Why would you need to raise a NotImplementedError(), is it due to you making sure that the system knows you need the method if you decide to subclass? Can't think of anything else. What if the user decides they only need acquire_step_and_glue and not the rest?

Reply:
These methods will never be accessed by the program if the corresponding mode is not valid, unless the user explicitly calls them. Hence, I believe it is reasonable to raise a non-implementation error. More on this on the new commit's DAQ class docstrings.